compiler/noirc_evaluator/src/ssa/opt/unrolling.rs

```diff
@@ -377,8 +380,7 @@ impl<'f> LoopIteration<'f> {
             }
         }
 
-        self.induction_value
-            .expect("Expected to find the induction variable by end of loop iteration")
+        self.induction_value.ok_or_else(CallStack::new)
```

Comment:
Should there be some custom error type to preserve this message?

Reply:
I think I'll make this pass have two modes (acir and brillig) so that it still panics on the acir case, but not panics on the brillig one.

Reply:
Done!